Code/Framework/AzCore/AzCore/Math/Sfmt.cpp

```diff
@@ -564,6 +553,8 @@ namespace AZ
     //=========================================================================
     void Sfmt::FillArray32(AZ::u32* array, int size)
     {
+        AZStd::lock_guard<decltype(m_generationMutex)> lock(m_generationMutex);
```

Comment:
Class template argument deduction can be used here to deduce the argument type.
```suggestion
        AZStd::scoped_lock lock(m_generationMutex);
```